discovery/kubernetes/pod.go

```diff
@@ -285,8 +296,10 @@ func (p *Pod) buildPod(pod *apiv1.Pod) *targetgroup.Group {
 		if len(c.Ports) == 0 {
 			// We don't have a port so we just set the address label to the pod IP.
 			// The user has to add a port manually.
+			// If pod.Status.PodIP is a IPv6 address, use JoinHostPort returns "[host]:port"
+			addr := net.JoinHostPort(pod.Status.PodIP, "80")
```

Comment:
Does this contradict the comment above? "just set the address label to the pod IP"
What happens if the URL is `https` ?
Seems like this would be a breaking change unexpected by many users.